module/zfs/arc.c

```diff
@@ -5948,6 +5949,13 @@ arc_read(zio_t *pio, spa_t *spa, const blkptr_t *bp,
 		 * Embedded BP's have no DVA and require no I/O to "read".
 		 * Create an anonymous arc buf to back it.
 		 */
+		blkptr_valid = zfs_blkptr_verify(spa, bp, zio_flags &
```

Comment:
Since `blkptr_valid` isn't used elsewhere we could just do the check in the conditional.